invokeai/backend/model_manager/install.py

```diff
@@ -0,0 +1,758 @@
+        directory.
+
+        May raise an UnknownModelException.
+        """
+        pass
+
+    @abstractmethod
+    def scan_models_directory(self):
+        """
+        Scan the models directory for new and missing models.
+
+        New models will be added to the storage backend. Missing models
+        will be deleted.
+        """
+        pass
+
+
+class ModelInstall(ModelInstallBase):
+    """Model installer class handles installation from a local path."""
+
+    _app_config: InvokeAIAppConfig
+    _logger: Logger
+    _store: ModelConfigStore
+    _download_queue: DownloadQueueBase
+    _async_installs: Dict[Union[str, Path, AnyHttpUrl], Optional[str]]
+    _installed: Set[str] = Field(default=set)
+    _tmpdir: Optional[tempfile.TemporaryDirectory]  # used for downloads
+    _cached_model_paths: Set[Path] = Field(default=set)  # used to speed up directory scanning
+
+    _legacy_configs: Dict[BaseModelType, Dict[ModelVariantType, Union[str, dict]]] = {
+        BaseModelType.StableDiffusion1: {
+            ModelVariantType.Normal: "v1-inference.yaml",
+            ModelVariantType.Inpaint: "v1-inpainting-inference.yaml",
+        },
+        BaseModelType.StableDiffusion2: {
+            ModelVariantType.Normal: {
+                SchedulerPredictionType.Epsilon: "v2-inference.yaml",
+                SchedulerPredictionType.VPrediction: "v2-inference-v.yaml",
+            },
+            ModelVariantType.Inpaint: {
+                SchedulerPredictionType.Epsilon: "v2-inpainting-inference.yaml",
+                SchedulerPredictionType.VPrediction: "v2-inpainting-inference-v.yaml",
+            },
+        },
+        BaseModelType.StableDiffusionXL: {
+            ModelVariantType.Normal: "sd_xl_base.yaml",
+        },
+        BaseModelType.StableDiffusionXLRefiner: {
+            ModelVariantType.Normal: "sd_xl_refiner.yaml",
+        },
+    }
+
+    def __init__(
+        self,
+        store: Optional[ModelConfigStore] = None,
+        config: Optional[InvokeAIAppConfig] = None,
+        logger: Optional[Logger] = None,
+        download: Optional[DownloadQueueBase] = None,
+        event_handlers: List[DownloadEventHandler] = [],
+    ):  # noqa D107 - use base class docstrings
+        self._app_config = config or InvokeAIAppConfig.get_config()
+        self._logger = logger or InvokeAILogger.get_logger(config=self._app_config)
+        self._store = store or get_config_store(self._app_config.model_conf_path)
+        self._download_queue = download or DownloadQueue(config=self._app_config, event_handlers=event_handlers)
+        self._async_installs: Dict[Union[str, Path, AnyHttpUrl], Union[str, None]] = dict()
+        self._installed = set()
+        self._tmpdir = None
+
+        # this step synchronizes the `models` directory with the models db
+        # do NOT do this automatically, but only on app startup
+        # self.scan_models_directory()
```

Comment:
You can implement a `start(self, invoker: Invoker)` method on a service and it will be called when the services (the invoker) inits.

Same for `stop(self)` - runs when the invoker stops.

Reply:
Thanks. Implemented.